manticore/core/smtlib/solver.py

```diff
@@ -78,6 +78,8 @@ class SolverType(config.ConfigEnum):
 )
 RE_MIN_MAX_OBJECTIVE_EXPR_VALUE = re.compile(r"(?P<expr>.*?)\s+\|->\s+(?P<value>.*)", re.DOTALL)
 
+SOLVER_STATS = dict({"unknown": 0, "timeout": 0})
```

Comment:
Manticore relies on multi-threads
https://github.com/trailofbits/manticore/blob/5a258f499098394c0af25e2e3f00b1b603c2334d/manticore/core/manticore.py#L383-L384

So if we share the dict among all the threads, we might suffer from race condition. 

I am not too familiar with the thread architecture, but I suppose we could use a dict with a lock, or create one dict per thread?

Reply:
For future reference, you can use the locked context in situations where you need to store data securely across multiple worker threads or processes:
https://github.com/trailofbits/manticore/blob/master/manticore/core/manticore.py#L992-L1021

Unfortunately, buried down in SMTLib as we are, it's a bit difficult to access the locked context from here. As long as the stats are informational only, I would also support storing them at the module level. The built-in types are inherently thread-safe (thanks to the GIL) so as long as we're running in multithreaded mode (the default) this should be fine. 

As a minor nit, `{"unknown": 0, "timeout": 0}` is already a dictionary; no need to explicitly wrap it with the call to `dict()`.

Reply:
I think using a global dictionary looks like a reasonable solution so far.

> As a minor nit, {"unknown": 0, "timeout": 0} is already a dictionary; no need to explicitly wrap it with the call to dict().

Fixed.